src/burst_buffer/burst_buffer.lua

```diff
@@ -1,39 +1,722 @@
+	slurm.log_info("%s: slurm_bb_job_process(). job_script=%s",
+		lua_script_name, job_script)
+
+	-- Note: In this version of Slurm we do not have the job ID in this
+	-- function; it's coming in a later version.  So we have no way to
+	-- name the Workflow so that it can be found in a later step.
+	-- In a future version of Slurm this function will also get a user ID
+	-- and group ID.
+	-- For now we will create the Workflow resource with a temporary name
+	-- and with placeholder values for the user ID and group ID.  We will
+	-- submit it, report on whether it was good, and then delete it.  The
+	-- slurm_bb_setup() stage will have to re-create it using the job ID to
+	-- name it.
+
+	local workflow_name = "temp-" .. math.random(10000)
+	local workflow = DWS(workflow_name)
+	local done, err = make_workflow(workflow, job_script, 1, 1, 1)
+	if done == false then
+		slurm.log_error("%s: slurm_bb_job_process(). %s", lua_script_name, err)
+		return slurm.ERROR, err
+	end
+
+	-- The job script's directives are good.
+	-- Now throw away this temporary Workflow resource.
+	-- In slurm_bb_setup() it'll be created again using the job ID for its
+	-- name so it can be found in all other stages.
+
+	done, err = workflow:delete()
+	if done == false then
+		slurm.log_error("%s: slurm_bb_job_process(). %s", lua_script_name, err)
+		return slurm.ERROR, err
+	end
+
+	return slurm.SUCCESS
+end
+
+
+--[[
+--slurm_bb_pools
+--
+--WARNING: This function is called from slurmctld and must return quickly.
+--
+--This function is called on slurmctld startup, and then periodically while
+--slurmctld is running.
+--
+--You may specify "pools" of resources here. If you specify pools, a job may
+--request a specific pool and the amount it wants from the pool. Slurm will
+--subtract the job's usage from the pool at slurm_bb_data_in and Slurm will
+--add the job's usage of those resources back to the pool after
+--slurm_bb_teardown.
+--A job may choose not to specify a pool even you pools are provided.
+--If pools are not returned here, Slurm does not track burst buffer resources
+--used by jobs.
+--
+--If pools are desired, they must be returned as the second return value
+--of this function. It must be a single JSON string representing the pools.
+--]]
+function slurm_bb_pools()
+
+	slurm.log_info("%s: slurm_bb_pools().", lua_script_name)
+
+	--This commented out code specifies pools in a file:
+	--local pools_file, pools
+	--pools_file = "/path/to/file"
+
+	--io.input(pools_file)
+	--pools = io.read("*all")
+	--slurm.log_info("Pools file:\n%s", pools)
+
+	--This specifies pools inline:
+	local pools
+	pools ="\
+{\
+\"pools\":\
+  [\
+    { \"id\":\"pool1\", \"quantity\":1000, \"granularity\":1024 },\
+    { \"id\":\"pool2\", \"quantity\":5, \"granularity\":2 },\
+    { \"id\":\"pool3\", \"quantity\":4, \"granularity\":1 },\
+    { \"id\":\"pool4\", \"quantity\":25000, \"granularity\":1 }\
+  ]\
+}"
+
+	return slurm.SUCCESS, pools
```

Comment:
Leave a comment here that we are ignoring the `slurm_bb_pools` function. We can also comment out the log message and replace the pool list with a placeholder pool.